public/app/features/browse-dashboards/components/RecentlyDeletedActions.tsx

```diff
@@ -29,20 +29,31 @@ export function RecentlyDeletedActions() {
       .map(([uid]) => uid);
   }, [selectedItemsState.dashboard]);
 
+  const dashboardOrigin: { [key: string]: string } = {};
+  if (searchState.result) {
+    const originalLocations = selectedDashboards.map((selectedDashboard) => {
+      const index = searchState.result.view.fields.uid.values.findIndex((e) => e === selectedDashboard);
+      return searchState.result?.view.fields.location.values[index];
```

Comment:
ok so first up, you could just do `return searchState.result?.view.fields.uid.values.find((e) => e === selectedDashboard);` here

the `if (searchState.result) {` _should_ ensure that `searchState.result` is always defined, so you shouldn't need to do `searchState.result?.`. but for some reason it's not working. i suspect because you're now in another closure due to the `.map`. it doesn't really matter, because of the `.find` the type of `originalLocations` will always be `(string | undefined)[]`

all that to say, you want:
```tsx
if (searchState.result) {
  const originalLocations = selectedDashboards.map((selectedDashboard) => {
    return searchState.result?.view.fields.uid.values.find((e) => e === selectedDashboard);
  });
```

Reply:
ahhh gotcha, serves me right for reviewing quickly on a friday 😄 ignore that bit then. rest should still apply tho 👍

Reply:
think maybe just:
```tsx
  if (searchState.result) {
    for (const dashboard of selectedDashboards) {
      const index = searchState.result.view.fields.uid.values.findIndex((e) => e === dashboard);
      const location = searchState.result.view.fields.location.values[index];
      if (location) {
        dashboardOrigin[dashboard] = location;
      }
    }
  }
```
ends up being cleaner?

Reply:
This makes sense now. I made it even cleaner.